### PR TITLE
PMM-6279 Allow show multiple metadata values.

### PR DESCRIPTION
**Pull request by JiriCtvrtka**
PMM-6279

Link to the Feature Build: SUBMODULES-0

If this PR adds or removes or alters one or more API endpoints, please review and add or update the relevant [API documents](https://github.com/percona/pmm/tree/main/docs/api) as well:

- [ ] API Docs updated

If this PR is related to some other PRs in this or other repositories, please provide links to those PRs:

- Links to related pull requests (optional).

### CODE REVIEW

File: qan-api2/models/metrics.go [hunk 1]
```diff
@@ -1120,26 +1116,63 @@ func (m *Metrics) GetSelectedQueryMetadata(ctx context.Context, periodStartFromS
...
-			&res.ServiceType,
-			&res.ServiceId,
-			&res.Environment,
-			&res.NodeId,
-			&res.NodeName,
-			&res.NodeType)
+		row := make([]any, len(columnNames))
+		for i := range row {
+			row[i] = &row[i]
+		}
```
Comment (artemgavrilov) [inline qan-api2/models/metrics.go:1129]: It seems we have some problem here.

File: qan-api2/models/metrics.go [hunk 2]
```diff
@@ -1120,26 +1116,63 @@ func (m *Metrics) GetSelectedQueryMetadata(ctx context.Context, periodStartFromS
...
+func valueIsInArray(value string, array []string) bool {
+	for _, v := range array {
+		if v != value {
+			continue
+		}
+
+		return true
+	}
+
+	return false
```
Comment (artemgavrilov) [inline qan-api2/models/metrics.go:1177]:
```suggestion
	for _, v := range array {
		if v == value {
			return true
		}
	}

	return false
```

**Review by artemgavrilov — CHANGES_REQUESTED** [review]

File: qan-api2/models/metrics.go (hunk 1 above)
Comment (JiriCtvrtka) [reply qan-api2/models/metrics.go:1129]: Without that I cannot read it into map. Any ideas how to handle this better?

Comment (artemgavrilov) [reply qan-api2/models/metrics.go:1129]: Aha, it seems that I got what you're trying to do. Try this, it should be easier to understand:
```suggestion
		row := make([]any, len(columnNames))
		for i := range columnNames {
			row[i] = new(sql.RawBytes)
		}
```

Comment (artemgavrilov) [reply qan-api2/models/metrics.go:1129]: Or `new(any)`, probably will work as well

Comment (JiriCtvrtka) [reply qan-api2/models/metrics.go:1129]: Nice thanks. I will test it next week right after national holidays.

File: qan-api2/models/metrics.go [hunk 3]
```diff
@@ -1120,26 +1116,61 @@ func (m *Metrics) GetSelectedQueryMetadata(ctx context.Context, periodStartFromS
 	}
 	defer rows.Close()
 
+	metadata := make(map[string][]string)
```
[the remark this answers is not in the document]
Comment (BupycHuk) [reply qan-api2/models/metrics.go:1119]: in that case we don't need to check if the value is exist

**Review by BupycHuk — CHANGES_REQUESTED** [review]
Suggested performance improvements

File: qan-api2/models/metrics.go (hunk 1 above)
Comment (JiriCtvrtka) [reply qan-api2/models/metrics.go]: In the end I used new(string).

**Comment by JiriCtvrtka** [comment]
All required changes should be addressed now.

**Review by artemgavrilov — APPROVED** [review]

**Review by BupycHuk — APPROVED** [review]
👍🏼

**Pull request merged**